repose-aggregator/components/filters/header-normalization/src/main/scala/org/openrepose/filters/headernormalization/HeaderNormalizationFilter.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2016 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+
+package org.openrepose.filters.headernormalization
+
+import java.net.URL
+import java.util.concurrent.TimeUnit
+import javax.inject.{Inject, Named}
+import javax.servlet._
+import javax.servlet.http.HttpServletRequest
+
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.commons.utils.servlet.http.HttpServletRequestWrapper
+import org.openrepose.core.filter.FilterConfigHelper
+import org.openrepose.core.filters.HeaderNormalization
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.core.services.reporting.metrics.MetricsService
+import org.openrepose.core.services.reporting.metrics.impl.MeterByCategorySum
+import org.openrepose.filters.headernormalization.config.{HttpMethod, HeaderNormalizationConfig}
+
+import scala.collection.JavaConverters._
+import scala.util.matching.Regex
+
+@Named
+class HeaderNormalizationFilter @Inject()(configurationService: ConfigurationService, metricsService: MetricsService)
+  extends Filter with UpdateListener[HeaderNormalizationConfig] with LazyLogging {
+
+  import HeaderNormalizationFilter._
+
+  private var configurationFile: String = DEFAULT_CONFIG
+  private var initialized = false
+  private var config: Seq[Target] = _
+  private var metricsMeter: Option[MeterByCategorySum] = _
+
+  override def init(filterConfig: FilterConfig): Unit = {
+    logger.trace("Header Normalization filter initializing...")
+    configurationFile = new FilterConfigHelper(filterConfig).getFilterConfig(DEFAULT_CONFIG)
+
+    logger.info(s"Initializing Header Normalization filter using config $configurationFile")
```

Comment:
Interesting.  I'm hoping it's a safe bet to use interpolation as the defacto method of concatenating strings that should continuously see improvements in terms of speed.  Just like how concatenating strings have been changed in the background to be faster in later versions of the JVM.